mermaid_classifier/pyspacer/train.py

```diff
@@ -1659,9 +1660,45 @@ def log_confusion_matrix(
             for bagf_id in bagf_ids_in_freq_order
         ]
 
-        # To dataframe, labeling each column with a BA-GF combo.
-        df = pd.DataFrame(data=matrix, columns=bagf_names)
-        # Add column to label each row with a BA-GF combo.
-        df.insert(loc=0, column='-', value=bagf_names)
-        # Log the confusion matrix as a table.
-        mlflow.log_table(df, filename)
+        # Create figure with size scaled to number of labels
+        num_labels = len(bagf_names)
+        fig_size = max(12, num_labels * 0.6)
+        fig, ax = plt.subplots(figsize=(fig_size, fig_size))
+
+        # Create confusion matrix display
+        display = sklearn.metrics.ConfusionMatrixDisplay(
+            confusion_matrix=matrix,
+            display_labels=bagf_names
+        )
+
+        # Plot with format string to avoid scientific notation
+        # 'd' for integers (frequencies), '.0f' for floats displayed as integers (percents)
+        display.plot(
+            ax=ax,
+            cmap='viridis',
+            values_format='d' if not normalize else '.0f'
```

Comment:
Or use `np.int64()` in the `normalize` case up above, so we can unconditionally use `'d'` here.